File: 1-algo-toolbox/1.1-challenges/gcd.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

rl.question("Enter two numbers: ", (input) => {
  let [a, b] = input.split(" ").map(Number);
  console.log(lcm(a, b));
  rl.close();
});

function gcd(a, b) {
  if (b == 0)
    return a;
  return gcd(b, a % b);
}

function lcm(a, b) {
  return (a / gcd(a, b)) * b;
}

File: 1-algo-toolbox/1.1-challenges/gcd.c
#include <stdio.h>

typedef long long ll;

int gcd(int a, int b) {
    if (b == 0)
        return a;
    return gcd(b, a % b);
}

ll lcm(int a, int b) {
    return ((ll) a / gcd(a, b)) * b;
}

int main() {
    int a, b;
    scanf("%d %d", &a, &b);
    printf("%lld\n", lcm(a, b));
    return 0;
}
